src/include/platform/ConfigurationManager.h

```diff
@@ -137,9 +137,14 @@ class ConfigurationManager
     friend class ::chip::DeviceLayer::Internal::GenericPlatformManagerImpl;
     template <class>
     friend class ::chip::DeviceLayer::Internal::GenericPlatformManagerImpl_POSIX;
+#ifdef __clang__
     // Parentheses used to fix clang parsing issue with these declarations
+    friend CHIP_ERROR(::chip::Platform::PersistedStorage::Read(::chip::Platform::PersistedStorage::Key key, uint32_t & value));
+    friend CHIP_ERROR(::chip ::Platform::PersistedStorage::Write(::chip::Platform::PersistedStorage::Key key, uint32_t value));
```

Comment:
if we use the CHIP_ERROR( syntax, would it work in GCC?

needing an ifdef sounds odd ... is C++ not standard? 


there seems to be an extra space in the namespace for Write

Reply:
@vivien-apple ?

Reply:
Oof. Can we make a macro to address portability issues like this?

Reply:
GCC/Clang will be both happy with parentheses. The issue being some zealous flags that will  complains with `ConfigurationManager.h:141:22: error: unnecessary parentheses in declaration of 'Read' [-Werror=parentheses]`

Reply:
I reworked the parentheses so CI is happy, Clang is happy and I hope you will be happy too :)